### PR TITLE
Remove references to manually use `withSentry` from Next.js docs

### PR DESCRIPTION
**Pull request by lforst**
Ref: https://github.com/getsentry/sentry-javascript/issues/5505

Removes all references and instructions to use `withSentry` from the docs, under the assumption that users will use our new auto-wrapping functionality of Next.js data fetching methods and API routes instead.

### CODE REVIEW

File: src/platforms/javascript/guides/nextjs/manual-setup.mdx [hunk 1]
```diff
@@ -242,6 +220,86 @@ module.exports = withSentryConfig(moduleExports);
...
+
+```javascript {filename:next.config.js}
+const moduleExports = {
+  sentry: {
+    autoInstrumentServerFunctions: false,
+  },
+};
+```
+
+Under the hood, when using this option, the SDK is utilizing a Webpack loader to wrap all your API route handlers and data fetching methods.
```
Comment (imatwawana) [inline src/platforms/javascript/guides/nextjs/manual-setup.mdx:237]:
```suggestion
Under the hood, when using this option, the SDK is using a Webpack loader to wrap all your API route handlers and data fetching methods.
```

**Review by imatwawana — COMMENTED** [review]
Made some styling edits

**Comment by lforst** [comment]
Sorry for tagging everybody 😬 Git be hard sometimes

File: src/platforms/javascript/guides/nextjs/manual-setup.mdx [hunk 2]
```diff
@@ -242,21 +219,53 @@ module.exports = withSentryConfig(moduleExports);
 
 In that case you can also skip the `sentry-cli` configuration step below.
 
-### Automatically Instrument API Routes and Data Fetching Methods
+### Disable Automatic Instrumentation of API Routes and Data Fetching Methods
```
Comment (AbhiPrasad) [inline src/platforms/javascript/guides/nextjs/manual-setup.mdx:222]:
```suggestion
### Automatic Instrumentation of API Routes and Data Fetching Methods
```

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**